libsolidity/ast/Types.cpp

```diff
@@ -1360,11 +1360,11 @@ string ArrayType::identifier() const
 	else
 	{
 		id = "t_array";
-		id += identifierList(baseType());
+		id += "_" + identifierList(baseType());
```

Comment:
Yeah, I think you should not even put a `_` here, because it would translate to `t_array,`, but it should be `t_array$_` which is `t_array(`.